marklogic-data-hub/src/main/java/com/marklogic/hub/impl/EntityManagerImpl.java

```diff
@@ -35,32 +35,44 @@
 import com.marklogic.hub.EntityManager;
 import com.marklogic.hub.HubConfig;
 import com.marklogic.hub.HubProject;
-import com.marklogic.hub.entity.*;
+import com.marklogic.hub.entity.DefinitionType;
+import com.marklogic.hub.entity.DefinitionsType;
+import com.marklogic.hub.entity.HubEntity;
+import com.marklogic.hub.entity.InfoType;
+import com.marklogic.hub.entity.ItemType;
+import com.marklogic.hub.entity.PropertyType;
 import com.marklogic.hub.error.EntityServicesGenerationException;
 import com.marklogic.hub.util.HubModuleManager;
 import org.apache.commons.io.FileUtils;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.core.io.FileSystemResource;
 import org.springframework.core.io.Resource;
 import org.springframework.stereotype.Component;
+import org.springframework.web.context.annotation.SessionScope;
 
+import javax.annotation.PostConstruct;
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.IOException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
 import java.util.stream.Collectors;
 
+@SessionScope
 @Component
 public class EntityManagerImpl extends LoggingObject implements EntityManager {
     private static final String ENTITY_FILE_EXTENSION = ".entity.json";
 
     @Autowired
     private HubConfig hubConfig;
 
-    @Autowired
-    private HubProject hubProject;
+    private HubProject hubProject = null;
```

Comment:
I recommend just ditching this, it only complicates things. Just change each reference to hubConfig.getHubProject().